### PR TITLE
Add Component, Assignment, Exam, ClassParticipation

### PR DESCRIPTION
**Pull request by TrungBui32**
Add abstract class Component and its children: Assignment, Exam, ClassParticipation
Solve issues #48

### CODE REVIEW

**Comment by ThienDuc3112** [comment]
Please change UML to reflect the now abstract class Component

Other than that, LGTM

**Comment by ThienDuc3112** [comment]
Should Override equals and toString in each of the child class and the component class as well, you can update the UML to reflect that

the rest LGTM

**Review by ThienDuc3112 — CHANGES_REQUESTED** [review]
Needed to add equals and toString method

**Review by ThienDuc3112 — APPROVED** [review]
LGTM, just need to make it pass the code style check

File: src/main/java/tutorlink/componentpackage/Exam.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package tutorlink.componentpackage;
+
+public class Exam extends Component{
+    public Exam(String name, double maxScore, double weight) {
+        super(name, maxScore, weight);
+    }
+    @Override
+    public String toString() {
+        return "Exam " + super.toString();
```
Comment (RCPilot1604) [inline src/main/java/tutorlink/componentpackage/Exam.java:9]: Maybe for this you would want to include some formatting to make it clear that "Exam" is the description (component) and the following string is the name: 

Eg. "Exam: CS2113 finals"

**Review by RCPilot1604 — APPROVED** [review]
Looks good to me

**Review by jinzihan2002 — APPROVED** [review]
LGTM

**Pull request merged**